### PR TITLE
Segmented copy

### PR DESCRIPTION
**Pull request by hkaiser**
This adds the implementation of the segmented `parallel::copy` algorithm. It includes a couple of refinements to the iterators as exposed by `hpx::vector`.

### CODE REVIEW

**Comment by hkaiser** [comment]
All tests pass: http://hermione.cct.lsu.edu/builders/hpx_gcc_4_9_boost_1_55_debian_x86_64_debug/builds/180

File: hpx/components/vector/vector_segmented_iterator.hpp [hunk 1]
```diff
@@ -228,25 +271,21 @@ namespace hpx
 
         void increment()
         {
-            HPX_ASSERT(data_);
             ++local_index_;
```
Comment (sithhell) [inline hpx/components/vector/vector_segmented_iterator.hpp]: Why are the asserts not needed anymore?

Comment (sithhell) [inline hpx/components/vector/vector_segmented_iterator.hpp]: Ok, it's not accessing data_, so i guess this is fine.

File: hpx/parallel/execution_policy.hpp [hunk 2]
```diff
@@ -27,6 +28,19 @@
...
+    struct task_execution_policy_tag
+    {
+        task_execution_policy_tag() {}
+    };
+    /// \endcond
+
+    /// The execution policy tag \a task can be used to create a execution
+    /// policy which forces the given algorithm to be executed in an
+    /// asynchronous way.
+    static task_execution_policy_tag const task;
```
Comment (sithhell) [inline hpx/parallel/execution_policy.hpp]: Can't we switch to constexpr for those things?

File: hpx/components/vector/vector_segmented_iterator.hpp (hunk 1 above)
Comment (hkaiser) [inline hpx/components/vector/vector_segmented_iterator.hpp]: This iterator should work remotely as well in which case `data_` will be zero.

File: hpx/parallel/execution_policy.hpp (hunk 2 above)
Comment (hkaiser) [inline hpx/parallel/execution_policy.hpp]: We could if all compilers were supporting it.

**Pull request merged**